src/Features/CSharp/Portable/ValidateFormatString/ValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+            if (invocationExpr.ArgumentList?.Arguments == null)
+            {
+                return;
+            }
+
+            SimpleNameSyntax name;
+            // When calling string.Format(...), Format will be of type MemberAccessExpressionSyntax 
+            if (invocationExpr.Expression is MemberAccessExpressionSyntax memberAccessExpr)
+            {
+                if (memberAccessExpr.Name?.ToString() != "Format")
+                {
+                    return;
+                }
+
+                name = memberAccessExpr.Name;
+            }
+            // When using static System.String and calling Format(...), Format will be of type IdentifierNameSyntax
+            else if (invocationExpr.Expression is IdentifierNameSyntax identifierNameSyntax)
+            {
+                if (identifierNameSyntax.ToString() != "Format")
+                {
+                    return;
+                }
+
+                name = identifierNameSyntax;
+            }
+            else
+            {
+                return;
+            }
+
+            var arguments = invocationExpr.ArgumentList.Arguments;
+            var numberOfArguments = arguments.Count;
+
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(name, context.CancellationToken);
+           
+            if (!TryGetFormatMethod(context, numberOfArguments, symbolInfo, out var method))
+            {
+                return;
+            }
+
+            var hasIFormatProvider =
+                method.Parameters[0].Type.ToString().Equals(typeof(IFormatProvider).FullName);
+
+            if (!TryGetFormatStringAndLocation(
+                arguments,
+                hasIFormatProvider,
+                out var formatString,
+                out var formatStringLocation))
+            {
+                return;
+            }
+
+            ValidateAndReportDiagnostic(context, hasIFormatProvider, arguments.Count, formatString, formatStringLocation);
+        }
+
+        private static bool TryGetFormatStringAndLocation(
+            SeparatedSyntaxList<ArgumentSyntax> arguments, 
+            bool hasIFormatProvider, 
+            out string formatString, 
+            out Location formatStringLocation)
+        {
+            formatString = null;
+            formatStringLocation = null;
+
+            var formatArgumentSyntax = GetFormatStringArgument(arguments, hasIFormatProvider);
+            if (formatArgumentSyntax == null)
+            {
+                return false;
+            }
+
+            if (!formatArgumentSyntax.Expression.IsKind(SyntaxKind.StringLiteralExpression))
+            {
+                return false;
+            }
+
+            var formatLiteralExpressionSyntax = (LiteralExpressionSyntax)formatArgumentSyntax.Expression;
+            formatString = formatLiteralExpressionSyntax.Token.ValueText;
+            formatStringLocation = formatArgumentSyntax.GetLocation();
```

Comment:
This means that the squiggle's span will include the name in a named argument case.  I would tend to think that using just the location of the `LiteralExpressionSyntax` would be better.